desktop/core/src/desktop/js/apps/notebook/app.js

```diff
@@ -953,7 +954,21 @@ huePubSub.subscribe('app.dom.loaded', app => {
         huePubSub.subscribe(
           'editor.upload.query',
           query_id => {
-            viewModel.selectedNotebook().snippets()[0].uploadQuery(query_id);
+            const useInternalHueOptimizerStorage = true;
+            if (useInternalHueOptimizerStorage) {
+              const snippet = viewModel.selectedNotebook().snippets()[0];
+              const dialect = snippet.dialect();
+              const statement = snippet.statement_raw();
+              sqlParserRepository.getAutocompleteParser(dialect).then(parser => {
+                const parseResults = parser.parseSql(statement, '');
+                $.post('/metadata/api/optimizer/upload/query', {
```

Comment:
There are API helper files in Hue that we should use for backen communication. Have a look at `desktop/core/src/desktop/js/api/apiUtils.js` It will help you with error handling etc.